### ISSUE TITLE
Question about occulusion.

### ISSUE THREAD

**Issue by Promethe-us** [ethz-vlg/mvtracker#5]
Thanks for your great work!
I try to implement MV Tracker on robotics grasp task (2 cameras views), I set 50 points of the first frame, but it's track performance is inaccurate.
Do you have any suggestions about this high occulusion scenes?

Query Points:
<img width="592" height="605" alt="Image" src="https://github.com/user-attachments/assets/f3200a35-564b-44e0-a5b1-2cb035d66ec8" />

Video: 
https://drive.google.com/file/d/1DLfV5c-5OIRRN4XhgSG3o6l0b9mvqLrn/view?usp=sharing

**Comment by Promethe-us** [comment]
VIdeo: https://drive.google.com/file/d/18G33vAsi2O8CiIt6kzlKp-33SYtMzfsM/view?usp=sharing

**Comment by Yefancy** [comment]
Side question: Are the intrinsics, extrinsics, and depth in your test obtained directly from hardware, or estimated using models like VGGT?
In my own tests, when I used VGGT’s estimated results, the tracking performance turned out to be quite poor.

**Comment by Promethe-us** [comment]
Thanks for your reply!

The intrinsics, extrinsics, and depth are loaded by simulator (IsaacLab). You think the problem mainly lies in (1) simulation data don't appear in your training set; or (2) Hand Occulusion?

I train this on robot data from script can solve this?

**Comment by Yefancy** [comment]
sry that I am not the author of this work. I just observed similar issues as your video, such as noticeable self-offset of tracked points (even when the object is static). I wanted to check whether there might be some common reasons for this. 
well, I think the data from simulator should be sufficiently accurate, so I suspect the issue may not be caused by imprecise ground-truth data.

**Comment by m43** [comment]
Hi @Promethe-us! Could you please share your input (rgbs, depths, intrs, extrs, and maybe those query points) and I take a look? E.g., you could save the input you gave to the model into a numpy `.npz` or a torch `.pth` file.


> You think the problem mainly lies in (1) simulation data don't appear in your training set; or (2) Hand Occulusion?

It's more likely the hand occlusion, or maybe the scene is not normalized well for the tracker which I could check. If you can add more cameras (e.g., having four instead of two) could improve the coverage and reduce the occlusion


> I train this on robot data from script can solve this?

Training/fine-tuning on the target domain could definitely help

**Comment by Promethe-us** [comment]
> Hi [@Promethe-us](https://github.com/Promethe-us)! Could you please share your input (rgbs, depths, intrs, extrs, and maybe those query points) and I take a look? E.g., you could save the input you gave to the model into a numpy `.npz` or a torch `.pth` file.
> 
> > You think the problem mainly lies in (1) simulation data don't appear in your training set; or (2) Hand Occulusion?
> 
> It's more likely the hand occlusion, or maybe the scene is not normalized well for the tracker which I could check. If you can add more cameras (e.g., having four instead of two) could improve the coverage and reduce the occlusion
> 
> > I train this on robot data from script can solve this?
> 
> Training/fine-tuning on the target domain could definitely help

You are so nice~!
I upload my data and infer script on: https://drive.google.com/drive/folders/1VTyU79Qc4TeXnf9FV0S7m1JU9P_h45Uo?usp=sharing.

Looking forward to your reply, do you think implemment sam2  the restrict point place can help this?
